gtsam/hybrid/tests/testDCGaussianMixtureFactor.cpp

```diff
@@ -29,19 +29,64 @@ using noiseModel::Isotropic;
 using symbol_shorthand::M;
 using symbol_shorthand::X;
 
-/* ****************************************************************************
- * Test elimination on a switching-like hybrid factor graph.
- */
+/* ************************************************************************* */
+// Check iterators of empty mixture.
 TEST(DCGaussianMixtureFactor, Constructor) {
   DCGaussianMixtureFactor factor;
-
-  // Check iterators
   DCGaussianMixtureFactor::const_iterator const_it = factor.begin();
   CHECK(const_it == factor.end());
   DCGaussianMixtureFactor::iterator it = factor.begin();
   CHECK(it == factor.end());
 }
 
+/* ************************************************************************* */
+// "Add" two mixture factors together.
+TEST(DCGaussianMixtureFactor, Sum) {
+  DiscreteKey m1(1, 2), m2(2, 3);
+
+  auto A1 = Matrix::Zero(2, 1);
+  auto A2 = Matrix::Zero(2, 2);
+  auto A3 = Matrix::Zero(2, 3);
+  auto b = Matrix::Zero(2, 1);
+  Vector2 sigmas;
+  sigmas << 1, 2;
+  auto model = noiseModel::Diagonal::Sigmas(sigmas, true);
+
+  auto f10 = boost::make_shared<JacobianFactor>(X(1), A1, X(2), A2, b);
+  auto f11 = boost::make_shared<JacobianFactor>(X(1), A1, X(2), A2, b);
+  auto f20 = boost::make_shared<JacobianFactor>(X(1), A1, X(3), A3, b);
+  auto f21 = boost::make_shared<JacobianFactor>(X(1), A1, X(3), A3, b);
+  auto f22 = boost::make_shared<JacobianFactor>(X(1), A1, X(3), A3, b);
+  std::vector<GaussianFactor::shared_ptr> factorsA{f10, f11};
+  std::vector<GaussianFactor::shared_ptr> factorsB{f20, f21, f22};
+
+  // TODO(Frank): why specify keys at all? And: keys in factor should be *all*
+  // keys, deviating from Kevin's scheme. Should we index DT on DiscreteKey?
+  // Design review!
```

Comment:
I'll create an issue for this so we can discuss. I am not sure what you mean by "specify keys at all", but yes we are generalizing this to all the discrete keys rather than a single discrete key.